Bug in deleteRelCatEntry

vikram18rock opened the pull request:
modified:   block_access.cpp

vikram18rock committed:
Bug in deleteRelCatEntry
modified:   block_access.cpp

vikram18rock commented:
The correction is required because irrespective of the relation the previous code keeps deleting 6 in place of no_of_attrs. So I updated this to no_of_attrs of deleting relation

clifordjoshy commented on block_access.cpp:
```diff
@@ -944,7 +944,9 @@ int deleteRelCatEntry(recId relcat_recid, Attribute relcat_rec[6]) {
 
-	int no_of_attrs = relcat_rec[1].nval;
+	getRecord(relcat_rec, 4, relcat_recid.slot);
+	int no_of_attrs = relcat_rec[2].nval;
```
Correct me if I'm wrong, but isn't `relcat_rec[2]` refer to the number of attributes in the relation. As I understand it, what we're trying to do here is get the number of attributes from the relation catalog and then subtracting that from the number of records in the attribute catalog.

vikram18rock replied:
Yes, That is what we are doing. We are getting the no. of attributes to be deleted from the attribute catalog's number of records from the relation's relation catalog entry.

vikram18rock replied:
And the `no_of_attrs` we need must be obtained from the no of attrs of the relation we delete. Which can be any number (2, 3, 4). whereas the no. of attributes of the relation catalog is always `6.`

vikram18rock replied:
I need to add one more change to this as well. I am setting the slot as unoccupied after getrecord is done. If not, the expected results are not coming. Please allow me to do this @clifordjoshy.

clifordjoshy commented:
Thanks for catching this! I've left a comment. Could you take a look before we merge?

vikram18rock committed:
- Typo in previous commit corrected to `relcat_rec[1]`
- The slot map update should be done after the getRecord to avoid error messages

clifordjoshy commented:
Looks good!
